go/obscuronode/host/host.go

```diff
@@ -538,27 +514,83 @@ func (a *Node) monitorBlocks() {
 
 	// only process blocks if the node is running
 	for atomic.LoadInt32(a.stopNodeInterrupt) == 0 {
-		latestBlkHeader := <-listener
+		blkHeader := <-listener
+
 		// don't process blocks if the node is stopping
 		if atomic.LoadInt32(a.stopNodeInterrupt) == 1 {
 			return
 		}
-		block, err := a.ethClient.BlockByHash(latestBlkHeader.Hash())
+
+		// ignore blocks if bootstrapping is happening
+		if atomic.LoadInt32(a.bootstrappingComplete) == 0 {
+			log.Trace("Node in bootstrap - ignoring block %s", blkHeader.Hash())
+			continue
+		}
+
+		block, err := a.ethClient.BlockByHash(blkHeader.Hash())
 		if err != nil {
-			log.Panic("could not fetch block for hash %s. Cause: %s", latestBlkHeader.Hash().String(), err)
+			log.Panic("could not fetch block for hash %s. Cause: %s", blkHeader.Hash().String(), err)
 		}
 		blockParent, err := a.ethClient.BlockByHash(block.ParentHash())
 		if err != nil {
 			log.Panic("could not fetch block's parent with hash %s. Cause: %s", block.ParentHash().String(), err)
 		}
 
-		nodecommon.LogWithID(a.shortID, "Received a new block b_%d(%d)",
-			obscurocommon.ShortHash(latestBlkHeader.Hash()),
-			latestBlkHeader.Number.Uint64())
+		nodecommon.LogWithID(
+			a.shortID,
+			"Received a new block b_%d(%d)",
+			obscurocommon.ShortHash(blkHeader.Hash()),
+			blkHeader.Number.Uint64(),
+		)
 		a.blockRPCCh <- blockAndParent{obscurocommon.EncodeBlock(block), obscurocommon.EncodeBlock(blockParent)}
 	}
 }
 
+func (a *Node) bootstrapNode() types.Block {
+	var err error
+	var nextBlk *types.Block
+
+	// build up from the genesis block
+	// todo update to bootstrap from the last block in storage
+	// todo the genesis block should be the block where the contract was deployed
+	currentBlock, err := a.ethClient.BlockByNumber(big.NewInt(0))
+	if err != nil {
+		panic(err)
+	}
+
+	nodecommon.LogWithID(a.shortID, "Started node bootstrap with block %d", currentBlock.NumberU64())
+
+	for startTime := time.Now(); ; {
```

Comment:
I don't love this. I find it confusing. Normally, the first part of a `for` loop is for initialising counters, etc., not for declaring other variables that you want.

Reply:
I usually use this to declare/initialize scoped vars. Changed it 👍